### PR TITLE
#161291006 Integrate Facebook Login

### PR DESCRIPTION
**Pull request by KvNGCzA**
#### What does this PR do?
Allows users to sign in or signup with their Facebook account

#### Description of Task to be completed?
- have the `/api/v1/auth/facebook` route working

#### How should this be manually tested?
- start the server with `npm run start`
- hit the http://localhost:3000/api/v1/auth/facebook on your browser

#### Any background context you want to provide?
n/a

#### What are the relevant pivotal tracker stories?
[#161291006](https://www.pivotaltracker.com/story/show/161291006)

#### Screenshots (if appropriate)
when you hit the http://localhost:3000/api/v1/auth/facebook
screen shot 2018-11-01 at 5 32 58 pm

it directs you to facebook
screen shot 2018-11-01 at 5 30 01 pm

then redirects back to our website
screen shot 2018-11-01 at 5 30 13 pm

#### Questions:
n/a

### CODE REVIEW

**Comment by Darthrighteous** [comment]
Good work on this, looks good overall. Is there a way to return the users facebook profile url? I think it'll be useful to save this in our table, it can be part of the user's profile.

**Comment by KvNGCzA** [comment]
> Good work on this, looks good overall. Is there a way to return the users facebook profile url? I think it'll be useful to save this in our table, it can be part of the user's profile.
> 
> Also, one of your commits doesn't follow the commit naming convention

Thanks. I will work on returning the Facebook URL and I have fixed the commit message.

File: mockdata/facebookMockData.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    givenName: 'Christopher',
+    familyName: 'Akanmu'
+  },
+  photos: [{
+    value: 'https://facebook.com/profilepicture'
+  }]
+};
+
+export const accessToken = 'khgsda6d8tyi3bu97bfepiohnfl2n3pf0hoql.23 fnkiu3g';
+export const refreshToken = 'khsgad63iy789r32vyf923fv23o0fig2o3f';
```
Comment (mrSerious) [inline mockdata/facebookMockData.js]: Nice work Chris, but shouldn't we have these tokens saved as environment variables?

Comment (KvNGCzA) [reply mockdata/facebookMockData.js]: @mrSerious Thanks. No, these are dummy strings usually returned to the callback function by facebook. They are not really private data.

File: routes/api/users.js [hunk 2]
```diff
@@ -26,4 +28,22 @@ router.post(
...
+router.get('/auth/facebook/success', (req, res) => res.status(200).json({
+  message: 'Successfully logged in with facebook!',
+  status: 200
+}));
+
+// facebook failure page
+router.get('/auth/facebook/failure', (req, res) => res.status(401).json({
+  message: 'Facebook login failed!',
+  status: 401
+}));
```
Comment (Darthrighteous) [inline routes/api/users.js]: Doesn't it make more sense to redirect the user the homepage on success(with this success message), and back to the login page on failure (with this failure message)? I reckon passport provides "Flash messages" also for this.
Why have you opted to create two new routes for success and failure?

Comment (mbilesanmi) [reply routes/api/users.js]: I think @Darthrighteous your comment will apply on the frontend.

**Pull request merged**